diff --git a/target/linux/ath79/image/generic.mk b/target/linux/ath79/image/generic.mk
index f264c14314..6ce76d58d9 100644
--- a/target/linux/ath79/image/generic.mk
+++ b/target/linux/ath79/image/generic.mk
@@ -729,6 +729,15 @@ define Device/comfast_cf-e314n-v2
 endef
 TARGET_DEVICES += comfast_cf-e314n-v2
 
+define Device/comfast_cf-e320n-v2
+  SOC := qca9531
+  DEVICE_VENDOR := COMFAST
+  DEVICE_MODEL := CF-E320N
+  DEVICE_VARIANT := v2
+  IMAGE_SIZE := 7936k
+endef
+TARGET_DEVICES += comfast_cf-e320n-v2
+
 define Device/comfast_cf-e375ac
   SOC := qca9563
   DEVICE_VENDOR := COMFAST
