diff --git a/target/linux/ath79/generic/base-files/etc/board.d/02_network b/target/linux/ath79/generic/base-files/etc/board.d/02_network
index 191138f521..b540ff766e 100644
--- a/target/linux/ath79/generic/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/generic/base-files/etc/board.d/02_network
@@ -225,6 +225,7 @@ ath79_setup_interfaces()
 	comfast,cf-e110n-v2|\
 	comfast,cf-e120a-v3|\
 	comfast,cf-e314n-v2|\
+	comfast,cf-e320n-v2|\
 	compex,wpj531-16m|\
 	openmesh,a40|\
 	openmesh,a60|\
@@ -242,7 +243,7 @@ ath79_setup_interfaces()
 	tplink,wbs510-v1|\
 	tplink,wbs510-v2|\
 	ubnt,routerstation)
-		ucidef_set_interfaces_lan_wan "eth1" "eth0"
+		ucidef_set_interfaces_lan_wan "eth0" "eth1"
 		;;
 	comfast,cf-e375ac)
 		ucidef_add_switch "switch0" \
